### PR TITLE
fix(router): return promise with error handler

### PR DESCRIPTION
**Pull request by btford**
See https://github.com/angular/angular/pull/2528\#discussion_r32493195

### CODE REVIEW

File: modules/angular2/src/router/router.ts [hunk 1]
```diff
@@ -128,12 +128,10 @@ export class Router {
                 ObservableWrapper.callNext(this._subject, matchedInstruction.accumulatedUrl);
               });
 
-      PromiseWrapper.catchError(result, (err) => {
+      return PromiseWrapper.catchError(result, (err) => {
         this._finishNavigating();
-        return err;
+        throw err;
```
Comment (vicb) [inline modules/angular2/src/router/router.ts]: test ?

Comment (btford) [inline modules/angular2/src/router/router.ts]: https://github.com/angular/angular/blob/master/modules/angular2/test/router/router_integration_spec.ts#L57

Comment (vicb) [inline modules/angular2/src/router/router.ts]: The behavior has changed, the test should have changed to reflect this ? _(hum unless the only difference would be in ordering the microtasks)_

Anyway I think the this was passing even before the previous updates, is the source issue really resolved ?

Comment (btford) [inline modules/angular2/src/router/router.ts]: > Anyway I think the this was passing even before the previous updates, is the source issue really resolved ?

Yes, the source issue was already resolved, but as you noted, the order of these microtasks would be unexpecte. I don't think we consider the order of these microtasks to be a public API worth having a separate test for. I'm not sure how best to test that anyway.

**Comment by btford** [comment]
Landed as bc798b182dab1dd71cf257b8e8fe50c650796096.

**Pull request closed without merging**